wdl2cwl/main.py

```diff
@@ -277,99 +282,117 @@ def load_wdl_workflow(self, obj: WDL.Tree.Workflow) -> cwl.Workflow:
-                    )
-
-                    if inp.id not in input_defaults:
-                        wf_step_inputs.append(
-                            cwl.WorkflowStepInput(
-                                id=inp.id, source=source_str, **extras
-                            )
-                        )
-                    else:
-                        wf_step_inputs.append(
-                            cwl.WorkflowStepInput(
-                                id=inp.id, default=source_str, **extras
-                            )
-                        )
-                wf_step_outputs = (
-                    [
-                        cwl.WorkflowStepOutput(id=output.name)
-                        for output in callee.outputs
-                    ]
-                    if callee.outputs
-                    else []
-                )
-                wf_step_run = self.load_wdl_objects(callee)
-                wf_step = cwl.WorkflowStep(
-                    wf_step_inputs,
-                    id=call.name,
-                    run=wf_step_run,
-                    out=wf_step_outputs,
-                )
-                wf_steps.append(wf_step)
+            with WDLSourceLine(body_part, ConversionException):
+                if isinstance(body_part, WDL.Tree.Call):
+                    wf_steps.append(self.get_workflow_call(body_part))
+                elif isinstance(body_part, WDL.Tree.Scatter):
+                    is_scatter_present = True
+                    wf_steps.extend(self.get_workflow_scatter(body_part))
+        requirements = [cwl.ScatterFeatureRequirement()] if is_scatter_present else None
 
         return cwl.Workflow(
             id=wf_name,
             cwlVersion="v1.2",
             doc=wf_description,
+            requirements=requirements,
             inputs=inputs,
             steps=wf_steps,
             outputs=outputs,
         )
 
+    def get_workflow_scatter(self, scatter: WDL.Tree.Scatter) -> List[cwl.WorkflowStep]:
+        """Get the CWL Workflow Step equivalent of a list of WDL Scatter Object."""
+        scatter_steps: List[cwl.WorkflowStep] = []
+        for body_obj in scatter.body:
+            if isinstance(body_obj, WDL.Tree.Call):
+                wp_step = self.get_workflow_call(body_obj)
+            wp_step.scatter = self.scatter_names.pop()
+            scatter_steps.append(wp_step)
+        return scatter_steps
+
+    def get_workflow_call(self, call: WDL.Tree.Call) -> cwl.WorkflowStep:
+        """Get the CWL Workflow Step equivalent of a WDL Call Object."""
+        callee = call.callee
+        if callee is None:
+            raise ConversionException("callee is None. This should not be possible.")
+        cwl_callee_inputs = self.get_cwl_task_inputs(callee.inputs)
+        call_inputs = call.inputs
+        inputs_from_call: Dict[str, Tuple[str, Dict[str, Any]]] = {}
+        input_defaults = set()
+        if call_inputs:
+            if_scatter_handled = False
```

Comment:
```suggestion
            scatter_handled = False
```
It is a little confusing to begin a variable with `if_`